pkg/provisioner/ironic/ironic.go

```diff
@@ -1119,7 +1119,7 @@ func (p *ironicProvisioner) Deprovision() (result provisioner.Result, err error)
 
 	switch nodes.ProvisionState(ironicNode.ProvisionState) {
 
-	case nodes.Error, nodes.CleanFail:
+	case nodes.Error, nodes.DeployFail:
 		if !ironicNode.Maintenance {
```

Comment:
I'm not sure why this was done, it doesn't seem to be needed in either case. Actually, automated cleaning will timeout and fail if maintenance is on.

Reply:
> 
> 
> > Yes, I would like to bring Ironic node from clean fail to manage in the event of deprovisioning
> 
> If we ever enable full cleaning, this ^^ will risk leaving user's data on disks. Since we don't use full cleaning, it's probably fine.

My use case is ironic node entered `clean fail` during provisioning, then we need to turn back to `manageable` with `manage` verb, not `deleted` verb as shown in deprovisoning code

Reply:
IMO, we need to take care not only `automated cleaning`, but also `manual cleaning` (for RAID and BIOS configuration)

Reply:
Right, but I'm not sure how it is related to my comment: any in-band clean steps will timeout and fail if in maintenance.

Reply:
This still needs addressing.